src/main/java/seedu/mtracker/filemanager/InstrumentEncoder.java

```diff
@@ -0,0 +1,23 @@
+package seedu.mtracker.filemanager;
+
+import seedu.mtracker.error.ErrorMessage;
+import seedu.mtracker.instrument.Instrument;
+
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+
+public class InstrumentEncoder {
+
+    public static void writeFile(ArrayList<Instrument> instruments, FileWriter writeToFile) throws IOException {
+        instruments.stream()
+                .forEach(instrument -> {
+                    try {
+                        writeToFile.write(instrument.textFileFormatting() + "\n");
+                    } catch (IOException e) {
+                        ErrorMessage.displayFileError();
+                    }
+                });
```

Comment:
can abstract this to a separate function
And the displayFileError I don't think is appropriate here. Should displayFileWriteError()
You may need to create specific displayFileErrors for different scenarios, loading the file, reading the file, writing the file etc